pkg/util/proto/document.go

```diff
@@ -219,27 +226,37 @@ func (d *Definitions) parseArbitrary(s *openapi_v2.Schema, path *Path) (Schema,
 // ParseSchema creates a walkable Schema from an openapi schema. While
 // this function is public, it doesn't leak through the interface.
 func (d *Definitions) ParseSchema(s *openapi_v2.Schema, path *Path) (Schema, error) {
+	if s.GetXRef() != "" {
+		return d.parseReference(s, path)
+	}
 	objectTypes := s.GetType().GetValue()
-	if len(objectTypes) == 1 {
+	if len(objectTypes) == 0 {
```

Comment:
We could avoid the repetition here I think, by extracting the `objectTypes[0]` into a variable, that we would default to `object`?

Reply:
I've replaced the if/else if with a switch. Not sure what you meant about extracting it into a variable - it wouldn't change anything as it's only used once.